libmikmod/loaders/load_amf.c

```diff
@@ -126,7 +126,16 @@ static BOOL AMF_UnpackTrack(MREADER* r)
 	/* read packed track */
 	if (r) {
 		tracksize=_mm_read_I_UWORD(r);
-		tracksize+=((ULONG)_mm_read_UBYTE(r))<<16;
+
+		/* The original code in DSMI library read the byte, but it is not used.
+		   Normally, it is zero, but Avoid.amf (version 8) have a track where its not.
+		   This track is garbage, so check the value and skip the track. */
+//		tracksize+=((ULONG)_mm_read_UBYTE(r))<<16;
+		if (_mm_read_UBYTE(r)!=0) {
+			_mm_fseek(modreader, tracksize*3, SEEK_CUR);
+			return 1;
+		}
+
```

Comment:
Judging from the handling in the DMP code, the time value of 0x70 is valid. The AMF loader loads the pattern data as-is. Here's the part of the playercode that parses the pattern data: https://github.com/ochrons/dsmi/blob/master/src/AMPLAYER.ASM#L1058. The affected track should reach that command value and then stop until the pattern ends. IMO sagamusix is correct.

GitHub won't let me highlight it, but I think this byte should be ignored and instead, in the check at lines 155-157, the remaining tracksize*3 bytes that haven't been read should be skipped (instead of returning an error). This would be more in line with the original player behavior.

All of the other changes look good to me. The only thing I found from the xmp report that isn't addressed here is the panning command. MikMod's handling of it looks unusual, but I can address it in a separate patch if needed once I find examples.

Reply:
I have new made the suggested change. The module can still be loaded and played.